tests/jballerina-unit-test/src/test/resources/test-src/types/xml/xml_iteration.bal

```diff
@@ -142,10 +140,103 @@ function xmlSequenceIter() returns string {
 
 function xmlCharItemIter() returns string {
     string result = "";
-
-    foreach xml|string elem in bitOfText {
+    int i = 0;
+    foreach xml elem in bitOfText {
         string str = io:sprintf("%s\n", elem);
         result += str;
+        i += 1;
     }
     return result;
 }
+
+function xmlTypeParamElementIter() {
```

Comment:
I think we can call these tests `testXmlElementSequenceIteration`, `testXmlElementSequenceIteration`, etc.